src/pickup.cpp

```diff
@@ -251,6 +251,116 @@ static bool select_autopickup_items( std::vector<item> &here, std::vector<bool>
     return bFoundSomething;
 }
 
+
+// parameter definition - append_fail_msg - any other message you want to append to the error message displayed
+//                      - silent_chk - if true, will not display any error message
+//                      - allow_swap - if true, when exceed volume, will always return true (unless current
+//                                     wielded weapon is no_unwield flag set, else return false
+//                      - chk_keep_hands_free - if true, will check for keep_hands_free. Note that this will not have any effect if not
+//                                               allow_swap
+//                      - chk_armor - if true, when item is armor and exceed volume, will return true as player might want to wear it
+bool Pickup::can_pick_one_up( player &p, item &newit, std::string append_fail_msg, bool silent_chk,
```

Comment:
It's probably good that this is in its own method, but it still looks like this set of checks (for weapon, worn, quiver, liquids, etc etc etc) is getting duplicated here.